main.tf

```diff
@@ -576,13 +576,13 @@ data "aws_iam_policy_document" "external_dns_irsa" {
 
 ## Crossplane Role IRSA
 module "iam_assumable_role_crossplane" {
-  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
-  version                       = "5.1.0"
-  create_role                   = true
-  role_name                     = "crossplane-${var.cluster_name}"
-  provider_url                  = aws_eks_cluster.cluster.identity[0].oidc[0].issuer
-  role_policy_arns              = [length(aws_iam_policy.crossplane) >= 1 ? aws_iam_policy.crossplane.arn : ""]
-  oidc_fully_qualified_subjects = ["system:serviceaccount:default:crossplane", "system:serviceaccount:db-controller:db-controller-db-controller"]
+  source                       = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                      = "5.1.0"
+  create_role                  = true
+  role_name                    = "crossplane-${var.cluster_name}"
+  provider_url                 = aws_eks_cluster.cluster.identity[0].oidc[0].issuer
+  role_policy_arns             = [length(aws_iam_policy.crossplane) >= 1 ? aws_iam_policy.crossplane.arn : ""]
+  oidc_subjects_with_wildcards = ["system:serviceaccount:default:bloxinabox-provider-*", "system:serviceaccount:db-controller:db-controller-db-controller"]
```

Comment:
Can we leave the db control on fully qualified and not in the wildcard section?

Reply:
Added a commit to address your request.